queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/alternate_name.py

```diff
@@ -116,16 +129,38 @@ def update_proprietor_change(
             raise DefaultException(f"No Proprietor in the SP {filing_type} for filing:{change_filing_rec.id}")
 
         proprietor, delivery_address, mailing_address = get_or_create_party(proprietor_dict, change_filing_rec)
+
         if not proprietor:
             raise DefaultException(f"No Proprietor in the SP {filing_type} for filing:{change_filing_rec.id}")
 
+    # Update operating name
+    name_request = dpath.util.get(change_filing, f"/{filing_type}/nameRequest", default=None)
+    identifier = dpath.util.get(change_filing_rec.filing_json, "filing/business/identifier")
+    if name_request and (to_legal_name := name_request.get("legalName")):
+        alternate_name = AlternateName.find_by_identifier(identifier)
+        proprietor = (
+            LegalEntity.find_by_id(alternate_name.legal_entity_id)
+            if alternate_name.legal_entity_id
+            else ColinEntity.find_by_id(alternate_name.colin_entity_id)
+        )
```

Comment:
do we need to retrieve `proprietor` here?  I think L131 retrieves it already?

Reply:
The filing json doesn't always have parties info if it doesn't plan to update it, so I retrieve proprietor here.